bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -621,7 +621,7 @@ private void removeListProcessChannel(List<Channel> removeList, String channelId
      * Remove previously statically created channels if the shade does not support them.
      */
     private void updateDynamicChannels(Capabilities capabilities) {
-        List<Channel> removeList = new ArrayList<Channel>(3);
+        List<Channel> removeList = new ArrayList<>();
```

Comment:
I would suggest to change this back to initialCapacity = 3. Otherwise 10 will be allocated, which is way more than the maximum of 3 needed.

Reply:
>  way more 

Wow! I dare note even think about the number of nano seconds that may be spared.. ;)

Seriously though: I changed it to the default because I had copy pasted the code (with your 3) to the tado binding, which requires 12, so that was was therefore wrong. And in such cases I think the preferred practice it to initialise lists with default settings and let Java sort things out automatically.